lib/asciidoctor-epub3/packager.rb

```diff
@@ -582,7 +582,16 @@ def package options = {}
         ::FileUtils.mkdir_p dest unless ::File.directory? dest
 
         epub_file = fmt == :kf8 ? %(#{::Asciidoctor::Helpers.rootname target}-kf8.epub) : target
-        builder.generate_epub epub_file
+
+        prev_zip_unicode = ::Zip.unicode_names
+        begin
+          # Let's hope there are no parallel threads running while we're messing with global variable
+          ::Zip.unicode_names = true
```

Comment:
If you're going to set it here, I would recommend just setting it globally. That way, at least it's only modified from the outside of calls to the APIs in this gem.

Ideally, this would be something that the Zip initializer would accept and that gepub would pass. But that could take awhile to see happen (always the risk of dependencies).

Reply:
> I would recommend just setting it globally

I'm not sure where I put this code.

Reply:
In the top-level require for Asciidoctor EPUB3, which would be at the end of lib/asciidoctor-epub3.rb.

Reply:
Okay. I'm a bit worried that we might affect other code that uses rubyzip, though hopefully this won't bring in any destructive effects.

Reply:
Better not to hide it. And it's not the only time we've done this (Asciidoctor PDF alters Prawn substantially).

Reply:
kk